packages/accounts/src/msca/plugins/multisig/middleware.ts

```diff
@@ -47,47 +72,96 @@ export const multisigSignatureMiddleware: ClientMiddlewareFn<{
     signer: account.getSigner(),
   });
 
-  // TODO: this needs to actually check the account's installed plugins and fetch the multisig plugin address
-  const threshold = await getThreshold(client, { account });
-
-  // if there is no override, then return the dummy signature
-  if (context?.signature == null) {
+  // then this is a propose operation
+  if (
+    context.userOpSignatureType === "UPPERLIMIT" &&
+    context?.signatures?.length == null &&
+    context?.aggregatedSignature == null
+  ) {
     return {
       ...resolvedStruct,
-      signature: await account.getDummySignature(),
+      signature: combineSignatures({
+        signatures: [
+          {
+            signature,
+            signer: await account.getSigner().getAddress(),
+            signerType,
+            userOpSigType: context.userOpSignatureType,
+          },
+        ],
+        upperLimitMaxFeePerGas: request.maxFeePerGas,
+        upperLimitMaxPriorityFeePerGas: request.maxPriorityFeePerGas,
+        upperLimitPvg: request.preVerificationGas,
+        usingMaxValues: false,
+      }),
     };
   }
 
-  if (!isHex(context.signature)) {
+  if (context.aggregatedSignature == null || context.signatures == null) {
     throw new InvalidContextSignatureError();
   }
 
+  // otherwise this is a sign operation
   const {
     upperLimitPvg,
     upperLimitMaxFeePerGas,
     upperLimitMaxPriorityFeePerGas,
-    signatures,
   } = await splitAggregatedSignature({
-    aggregatedSignature: context.signature,
-    threshold: Number(threshold),
+    aggregatedSignature: context.aggregatedSignature,
+    threshold: context.signatures.length + 1,
     account,
     request,
   });
 
   const finalSignature = combineSignatures({
-    signatures: signatures.concat({
-      userOpSigType: "ACTUAL",
+    signatures: context.signatures.concat({
+      userOpSigType: context.userOpSignatureType,
       signerType,
       signature,
       signer: await account.getSigner().getAddress(),
     }),
     upperLimitPvg,
     upperLimitMaxFeePerGas,
     upperLimitMaxPriorityFeePerGas,
+    usingMaxValues: isUsingMaxValues(request, {
+      upperLimitPvg,
+      upperLimitMaxFeePerGas,
+      upperLimitMaxPriorityFeePerGas,
+    }),
   });
 
   return {
     ...resolvedStruct,
     signature: finalSignature,
   };
 };
+
+const isUsingMaxValues = (
+  request: UserOperationRequest_v6 | UserOperationRequest_v7,
```

Comment:
you can just do `UserOperationRequest`: this is same thing as union type

with this pr https://github.com/alchemyplatform/aa-sdk/pull/607